### PR TITLE
Support specifying a root directory for a MarkLogic mount

### LINKED ISSUE #1766: Support specifying a root directory in MarkLogic mount config
The root directory will restrict operations on the mount to the specified directory and its descendants, akin to [`chroot`](https://en.wikipedia.org/wiki/Chroot) on *nix systems.

### PR DESCRIPTION
**Pull request by wemrysi**
Allows for specifying a physical path in the MarkLogic server when describing a mount to serve as the root directory for that mount, all operations and queries will be scoped within the specified directory.

Closes #1766.

### CODE REVIEW

**Comment by wemrysi** [comment]
@drostron This depends on #1850 so all those changes so up here as well, feel free to ignore everything other than the final commit.

File: marklogic/src/main/scala/quasar/physical/marklogic/fs/MarkLogicConfig.scala [hunk 1]
```diff
@@ -17,15 +17,23 @@
...
 import scalaz._, Scalaz._
 
 @Lenses
-final case class MarkLogicConfig(xccUri: URI, docType: MarkLogicConfig.DocType)
+final case class MarkLogicConfig(
+  xccUri: URI,
+  rootDir: ADir,
+  docType: MarkLogicConfig.DocType
+) {
+  def database: String = xccUri.getPath.drop(1)
```
Comment (drostron) [inline marklogic/src/main/scala/quasar/physical/marklogic/fs/MarkLogicConfig.scala]: Is this used anywhere?

**Comment by drostron** [comment]
LGTM. Just a quick question and I'll take another look once #1850 is in.

File: marklogic/src/main/scala/quasar/physical/marklogic/fs/MarkLogicConfig.scala (hunk 1 above)
Comment (wemrysi) [inline marklogic/src/main/scala/quasar/physical/marklogic/fs/MarkLogicConfig.scala]: Nope, removed.

**Comment by wemrysi** [comment]
@drostron Updated now that #1850 has been merged.

**Pull request merged**